include/FreeRTOS_IP_Private.h

```diff
@@ -418,6 +415,27 @@
 
     #endif /* ipconfigBYTE_ORDER == pdFREERTOS_LITTLE_ENDIAN */
 
+    #if ( ipconfigETHERNET_DRIVER_FILTERS_PACKETS == 0 )
+        #if ( ipconfigBYTE_ORDER == pdFREERTOS_LITTLE_ENDIAN )
+            /* The bits in the two byte IP header field that make up the fragment offset value. */
+            #define ipFRAGMENT_OFFSET_BIT_MASK         ( ( uint16_t ) 0xff0f )
+            /* The bits in the two byte IP header field that make up the flags value. */
+            #define ipFRAGMENT_FLAGS_BIT_MASK          ( ( uint16_t ) 0x00f0 )
+            /* Don't Fragment Flag */
+            #define ipFRAGMENT_FLAGS_DONT_FRAGMENT     ( ( uint16_t ) 0x0040 )
+            /* More Fragments Flag */
+            #define ipFRAGMENT_FLAGS_MORE_FRAGMENTS    ( ( uint16_t ) 0x0020 )
+        #else
+            /* The bits in the two byte IP header field that make up the fragment offset value. */
+            #define ipFRAGMENT_OFFSET_BIT_MASK         ( ( uint16_t ) 0x0fff )
```

Comment:
According to [this](https://tools.ietf.org/html/rfc791#page-11) RFC, the flags field is 3 bits long and the offset field is 13 bits long. Would it be more proper to make this `0x1fff` instead of what it is now?

Consider a scenario where the value of these 2 bytes in an incoming IP packet is `0x1000` (this would be the last fragment since the MF flag is not set and the offset is 0x1000 octets - should not be supported), it would pass the check you have modified in `FreeRTOS_IP.c` with the flags as they are now.

```
if( ( ( pxIPHeader->usFragmentOffset & ipFRAGMENT_OFFSET_BIT_MASK ) != 0U ) || ( ( pxIPHeader->usFragmentOffset & ipFRAGMENT_FLAGS_MORE_FRAGMENTS ) != 0U ) )
```

Reply:
You are absolutely right. I totally missed that. the offset mask is definitely 0x1FFF and 0xFF1F respectively.